src/Transports/MassTransit.EventHubIntegration/EventHubIntegration/EventHubProducerSendTransportContext.cs

```diff
@@ -1,48 +1,171 @@
 namespace MassTransit.EventHubIntegration
 {
     using System;
+    using System.Diagnostics;
     using System.Threading;
     using System.Threading.Tasks;
-    using Configuration;
+    using Azure.Messaging.EventHubs;
+    using Azure.Messaging.EventHubs.Producer;
     using MassTransit.Configuration;
     using Transports;
-    using Util;
 
 
     class EventHubProducerSendTransportContext :
         BaseSendTransportContext,
         EventHubSendTransportContext
     {
         readonly IHostConfiguration _configuration;
-        readonly Recycle<IProducerContextSupervisor> _producerContextSupervisor;
+        readonly EventHubEndpointAddress _endpointAddress;
+        readonly ISendPipe _sendPipe;
+        readonly IProducerContextSupervisor _supervisor;
 
-        public EventHubProducerSendTransportContext(IEventHubHostConfiguration hostConfiguration, ISendPipe sendPipe,
-            IHostConfiguration configuration, Uri endpointAddress, ISerialization serialization)
+        public EventHubProducerSendTransportContext(IProducerContextSupervisor supervisor, ISendPipe sendPipe,
+            IHostConfiguration configuration, EventHubEndpointAddress endpointAddress, ISerialization serialization)
             : base(configuration, serialization)
         {
+            _sendPipe = sendPipe;
+            _supervisor = supervisor;
             _configuration = configuration;
-            SendPipe = sendPipe;
+            _endpointAddress = endpointAddress;
+        }
+
+        public async Task<EventHubSendContext<T>> CreateContext<T>(T value, IPipe<EventHubSendContext<T>> pipe, CancellationToken cancellationToken,
+            IPipe<SendContext<T>> initializerPipe = null)
+            where T : class
+        {
+            var context = new EventHubMessageSendContext<T>(value, cancellationToken)
+            {
+                Serializer = Serialization.GetMessageSerializer(),
+                DestinationAddress = _endpointAddress
+            };
+
+            if (pipe is ISendPipe sendPipe)
+                await sendPipe.Send(context).ConfigureAwait(false);
+
+            await _sendPipe.Send(context).ConfigureAwait(false);
```

Comment:
I have my doubts about ordering of pipes here, can you please take a look @phatboyg ?